crypto/sm2/sm2_aid.c

```diff
@@ -0,0 +1,45 @@
+/*
```

Comment:
It seems this whole file is not needed anymore, since the new DER part has already handled this.

Reply:
The infrastructure is there, but you will need to add SM2 specific.

Reply:
Yes. SM2 specific stuffs are added in the latest commit - hope I have done right...

Reply:
I can help with those details this evening

Reply:
Thanks! One struggling thing is SM2 has no PublicKey OID. An SM2 key is encoded with `id_ecPublickey` IIRC at current stage, which is compatible with OpenSSL 1.1.1.

Reply:
Yes, I know. That is a bit of a problem, and means that certain assumptions need to be made...
I will have to look again how I solved that for the legacy implementation. I forget...